### ISSUE TITLE
Better errors for WinPTY

### ISSUE THREAD

**Issue by chrisduerr** [jwilm/alacritty#2344]
In the https://github.com/jwilm/alacritty/pull/2261 PR the ConPTY errors have been improved to make it obvious why a shell could not be started. However, the WinPTY messages are unchanged and there currently is no output at all when spawning the WinPTY failed.

It should be made clear to the user that the shell could not be found when starting Alacritty with an invalid shell.

Based on the little research I've done, it seems like there's no place in Alacritty to catch this error and it's swallowed further down in the WinPTY dependency chain. Maybe @zacps knows something about this, but I'd assume that we might have to fix something in the `winpty-sys` crate.

**Comment by cup** [comment]
https://github.com/jwilm/alacritty/issues/2016

**Comment by davidhewitt** [comment]
Given I've been refactoring winpty stuff already in #3054, want me to see if I can resolve this as part of the PR?

**Comment by chrisduerr** [comment]
> Given I've been refactoring winpty stuff already in #3054, want me to see if I can resolve this as part of the PR?

I think having it as a separate PR would be best, for tracking the issue nicely and making things easier to review without having PRs around for centuries. But if you think there's an issue solution to this that is trivial, I don't mind.

**Comment by davidhewitt** [comment]
> I think having it as a separate PR would be best

Sure thing. In which case I'll probably have a go some time after #3054 merges or closes, so that I don't have two in-flight PRs attempting to refactor the same code :smile:

**Comment by cup** [comment]
https://github.com/jwilm/alacritty/pull/3080
